scripts/calculate_percentile_offline.py

```diff
@@ -1,162 +1,140 @@
-
-
-def list_years_per_station():
-    """ List data years per station.
-
-    Returns:
-        Series: pandas dataframe, which you can kind of think of as a dictionary containing a list of
-        integers.
-    """
-    print('list_years_per_station...')
-    data_years = DAILY_WEATHER_DATA.groupby('station_code').year.unique()
-    data_years.apply(lambda x: x.sort())
-    return data_years
+    outside_month = (df['month'] < season['start_month']) | (
+        df['month'] > season['end_month'])
+    df.drop(df[outside_month].index, inplace=True)
 
+    before_start = ((df['month'] == season['start_month']) &
+                    (df['day'] < season['start_day']))
+    df.drop(df[before_start].index, inplace=True)
 
-def write_output_to_json(date_range, weather_stations, data_years, percentiles):
-    """ Write output to json file. """
-    print('write_output_to_json...')
-    # create season instance to be used in output
-    season = {
-        'start_month': FIRE_SEASON_START_MONTH,
-        'start_day': FIRE_SEASON_START_DATE,
-        'end_month': FIRE_SEASON_END_MONTH,
-        'end_day': FIRE_SEASON_END_DATE
-    }
-    output_folder = "app/data/{}-{}".format(date_range[0], date_range[1])
-    for weath_stat in weather_stations:
-        try:
-            key = weath_stat['code']
-            output_filename = output_folder + "/" + key + ".json"
-            station_summary = {
-                'ffmc': percentiles['ffmc'][int(key)],
-                'isi': percentiles['isi'][int(key)],
-                'bui': percentiles['bui'][int(key)],
-                'season': season,
-                'years': [int(year) for year in data_years[int(key)]],
-                'station': {
-                    'code': key,
-                    'name': weath_stat['name'],
-                    'lat': weath_stat['lat'],
-                    'long': weath_stat['long']
-                }
-            }
-            if not os.path.exists(output_folder):
-                os.mkdir(output_folder)
-            with open(output_filename, 'w+') as json_file:
-                json.dump(station_summary, json_file, indent=4, allow_nan=False)
-        except KeyError:
-            print('Data not available for {}'.format(key))
-            if os.path.exists(output_filename):
-                os.remove(output_filename)
-        except ValueError as e:
-            print('Value error for {} - {}'.format(key, e))
-            if os.path.exists(output_filename):
-                os.remove(output_filename)
+    after_end = ((df['month'] == season['end_month']) &
+                 (df['day'] > season['end_day']))
+    df.drop(df[after_end].index, inplace=True)
+
+
+def calculate_percentile(df, percentile: float) -> dict:
+    """ Calculate percentile """
+    ffmc = df[df['ffmc_valid']].ffmc.quantile(percentile)
+    bui = df[df['bui_valid']].bui.quantile(percentile)
+    isi = df[df['isi_valid']].isi.quantile(percentile)
+
+    ffmc = ffmc if not pd.isna(ffmc) else None
```

Comment:
I want to suggest that we round to 5 (or some other reasonable number of) digits after the decimal point to get rid of the floating point discrepancies we're getting from pandas.
```suggestion
    ffmc = round(ffmc, 5) if not pd.isna(ffmc) else None
```